src/screens/IndexScreen.js

```diff
@@ -374,223 +444,150 @@ export const IndexScreen = ({ navigation, route }) => {
+                  />
+                )}
+                {query === QUERY_TYPES.CATEGORIES && !!categoryListIntroText && (
+                  <Wrapper>
+                    <RegularText>{categoryListIntroText}</RegularText>
+                  </Wrapper>
+                )}
+              </>
+            }
+            ListEmptyComponent={
+              loading ? (
+                <LoadingContainer>
+                  <ActivityIndicator color={colors.accent} />
+                </LoadingContainer>
+              ) : showCalendar ? (
+                <Calendar
                   query={query}
                   queryVariables={queryVariables}
-                  fetchMoreData={isConnected ? fetchMoreData : null}
-                  refreshControl={
-                    <RefreshControl
-                      refreshing={refreshing}
-                      onRefresh={() => refresh(refetch)}
-                      colors={[colors.accent]}
-                      tintColor={colors.accent}
-                    />
-                  }
-                  showBackToTop
+                  calendarData={buildListItems(data, additionalData)}
+                  isLoading={loading}
+                  navigation={navigation}
+                />
+              ) : (
+                <EmptyMessage
+                  title={categories?.length ? texts.empty.categoryList : texts.empty.list}
+                  showIcon={!categories?.length}
                 />
+              )
+            }
+            ListFooterComponent={
+              <>
+                {query === QUERY_TYPES.CATEGORIES && !!categoryListFooter && (
+                  <>
+                    {!!categoryListFooter.footerText && (
+                      <Wrapper>
+                        <RegularText small>{categoryListFooter.footerText}</RegularText>
+                      </Wrapper>
+                    )}
+                    {!!categoryListFooter.url && !!categoryListFooter.buttonTitle && (
+                      <Wrapper>
+                        <Button
+                          onPress={() => openLink(categoryListFooter.url, openWebScreen)}
+                          title={categoryListFooter.buttonTitle}
+                        />
+                      </Wrapper>
+                    )}
+                  </>
+                )}
               </>
-            );
-          }}
-        </Query>
+            }
+            navigation={navigation}
+            data={
+              loading || (isCalendar && showCalendar) ? [] : buildListItems(data, additionalData)
+            }
+            horizontal={false}
+            sectionByDate={isCalendar ? !showCalendar : true}
+            query={query}
+            queryVariables={queryVariables}
+            fetchMoreData={isConnected ? fetchMoreData : null}
+            refreshControl={
+              <RefreshControl
+                refreshing={refreshing}
+                onRefresh={() => refresh(refetch)}
```

Comment:
as `refetch` comes from hook `const { data, loading, fetchMore, refetch } = useQuery(` now, we do not need to pass it to the `refresh` method, at is already defined.